agents/PCS-Large-Swap/src/agent.spec.ts

```diff
@@ -0,0 +1,202 @@
+import { TestTransactionEvent, createAddress, MockEthersProvider } from "forta-agent-tools/lib/tests";
+import { getPancakePairCreate2Address } from "./utils";
+import { ERC20ABI, PANCAKE_PAIR_ABI } from "./constants";
+import { BigNumber } from "ethers";
+import DataFetcher from "./data.fetcher";
+
+const createFinding = (
+  pairAddress: string,
+  swapTokenIn: string,
+  swapTokenOut: string,
+  swapAmountIn: BigNumber,
+  swapAmountOut: BigNumber,
+  percentageTokenIn: BigNumber,
+  percentageTokenOut: BigNumber,
+  swap_recipient: string
+): Finding => {
+  return Finding.from({
+    name: "Large Swap in stkBNB-BNB Pancakeswap pool",
+    description: "A swap that involved a significant percentage of a pool's liquidity was detected",
+    alertId: "pSTAKE-stkBNB-PCS-SUBSTANTIAL-Pool-Swap",
+    protocol: "stkBNB",
+    type: FindingType.Info,
+    severity: FindingSeverity.Info,
+    metadata: {
+      pancakePair: pairAddress,
+      tokenIn: swapTokenIn,
+      tokenOut: swapTokenOut,
+      amountIn: swapAmountIn.toString(),
+      amountOut: swapAmountOut.toString(),
+      percentageIn: percentageTokenIn.toString(),
+      percentageOut: percentageTokenOut.toString(),
+      swapRecipient: swap_recipient,
+    },
+  });
+};
+
+const PAIR_IFACE = new ethers.utils.Interface(PANCAKE_PAIR_ABI);
+const TOKEN_IFACE = new ethers.utils.Interface(ERC20ABI);
+const TEST_PANCAKE_FACTORY = createAddress("0x32");
+const [token0, token1, token2, token3] = [
+  createAddress("0x01"),
+  createAddress("0x02"),
+  createAddress("0x03"),
+  createAddress("0x04"),
+];
+const INIT_CODE = ethers.utils.keccak256("0x");
+const TEST_PAIR_ADDRESS = getPancakePairCreate2Address(TEST_PANCAKE_FACTORY, token0, token1, INIT_CODE).toLowerCase();
+const TEST_LARGE_THRESHOLD = BigNumber.from("10"); // percent
+export const SWAP_EVENT =
+  "event Swap(address indexed sender, uint amount0In, uint amount1In, uint amount0Out,uint amount1Out,address indexed to)";
+const SENDER = createAddress("0x12");
+
+const toEbn = (num: string) => ethers.BigNumber.from(num);
+
+const createSwapEvent = (
+  pairAddress: string,
+  amount0In: ethers.BigNumber,
+  amount1In: ethers.BigNumber,
+  amount0Out: ethers.BigNumber,
+  amount1Out: ethers.BigNumber,
+  to: string
+): [string, string, string, string, string] => {
+  const data = ethers.utils.defaultAbiCoder.encode(
+    ["uint", "uint", "uint", "uint"],
+    [amount0In, amount1In, amount0Out, amount1Out]
+  );
+  const senderTopic = encodeParameter("address", SENDER);
+  const toTopic = encodeParameter("address", to);
+  return ["Swap(address,uint256,uint256,uint256,uint256,address)", pairAddress, data, senderTopic, toTopic];
+};
+
+describe("PancakeSwap Large Swap Bot Test Suite", () => {
+  const mockProvider: MockEthersProvider = new MockEthersProvider();
+  const mockFetcher: DataFetcher = new DataFetcher(mockProvider as any);
+  const handleTransaction: HandleTransaction = provideBotHandler(
+    TEST_LARGE_THRESHOLD,
```

Comment:
100% swap not possible since cannot swap entire pool as it is x*y=k. hence did not cover.

Reply:
Can you add a comment over LARGE_THRESHOLD on the range of value we can put, and why we cannot provide 100%, This will help the programmer who will be changing the values.